js/src/api/rpc/parity/parity.js

```diff
@@ -374,9 +374,21 @@ export default class Parity {
       .execute('parity_postSign', inAddress(address), inHex(hash));
   }
 
-  postTransaction (options) {
+  postTransaction (_options = {}) {
+    const { metadata = {}, ...options } = _options;
+    const parsedOptions = inOptions(options);
+
     return this._transport
-      .execute('parity_postTransaction', inOptions(options));
+      .execute('parity_postTransaction', parsedOptions)
+      .then((requestId) => {
+        this._transport.emit('request', {
```

Comment:
This would make more sense on the logging layer where events are already fired. As it stands it now does the "same" thing in 2 different places.